jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletCoreResponse.java

```diff
@@ -0,0 +1,378 @@
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.IO;
+
+/**
+ * A {@link HttpServletResponse} wrapped as a core {@link Response}.
+ */
+class ServletCoreResponse implements Response
+{
+    private final HttpServletResponse _response;
+    private final Request _coreRequest;
+    private final HttpFields.Mutable _httpFields;
+    private final boolean _included;
+    private final ServletContextResponse _servletContextResponse;
+
+    public ServletCoreResponse(Request coreRequest, HttpServletResponse response, boolean included)
+    {
+        _coreRequest = coreRequest;
+        _response = response;
+        _servletContextResponse = ServletContextResponse.getServletContextResponse(response);
+        HttpFields.Mutable fields = new HttpServletResponseHttpFields(response);
+        if (included)
+        {
+            // If included, accept but ignore mutations.
+            fields = new HttpFields.Mutable.Wrapper(fields)
+            {
+                @Override
+                public HttpField onAddField(HttpField field)
+                {
+                    return null;
+                }
+
+                @Override
+                public boolean onRemoveField(HttpField field)
+                {
+                    return false;
+                }
+            };
+        }
+        _httpFields = fields;
+        _included = included;
+    }
+
+    @Override
+    public HttpFields.Mutable getHeaders()
+    {
+        return _httpFields;
+    }
+
+    public HttpServletResponse getServletResponse()
+    {
+        return _response;
+    }
+
+    @Override
+    public boolean isLast()
+    {
+        return _servletContextResponse.isLast();
+    }
+
+    @Override
+    public boolean isCompletedSuccessfully()
+    {
+        return _servletContextResponse.isCompletedSuccessfully();
+    }
+
+    @Override
+    public boolean isCommitted()
+    {
+        return _response.isCommitted();
+    }
+
+    public boolean isWriting()
+    {
+        return _servletContextResponse.isWriting();
+    }
+
+    @Override
+    public void write(boolean last, ByteBuffer byteBuffer, Callback callback)
+    {
+        if (_included)
+            last = false;
+        try
+        {
+            if (BufferUtil.hasContent(byteBuffer))
+            {
+                if (isWriting())
+                {
+                    String characterEncoding = _response.getCharacterEncoding();
+                    try (ByteBufferInputStream bbis = new ByteBufferInputStream(byteBuffer);
+                         InputStreamReader reader = new InputStreamReader(bbis, characterEncoding))
+                    {
+                        IO.copy(reader, _response.getWriter());
+                    }
+
+                    if (last)
+                        _response.getWriter().close();
+                }
+                else
+                {
+                    BufferUtil.writeTo(byteBuffer, _response.getOutputStream());
```

Comment:
Theoretically it might be possible to create a wrapper that could use async servlet writes (if async has been started), but currently I am not sure it is worth the hassle.... as currently this is only used if the response is wrapped and wrapping servlet async is next to impossible.  I'll add javadoc